File: python/hash/884. 两句话中的不常见单词.py
from typing import List
class Solution:
    def uncommonFromSentences(self, A: str, B: str) -> List[str]:
        from collections import defaultdict
        d1 = defaultdict(int)
        d2 = defaultdict(int)

        for word in A.split(" "):
            d1[word]+=1
        for word in B.split(" "):
            d2[word]+=1

        res = []
        for k,v in d1.items():
            if v == 1 and k not in d2:
                res.append(k)

        for k,v in d2.items():
            if v == 1 and k not in d1:
                res.append(k)
        return res

File: go/hash/884. 两句话中的不常见单词.go
package main

import "strings"

func uncommonFromSentences(A string, B string) []string {
	d1, d2 := make(map[string]int), make(map[string]int)

	tmpA := strings.Split(A, " ")
	tmpB := strings.Split(B, " ")

	for _, v := range tmpA {
		d1[v]++
	}

	for _, v := range tmpB {
		d2[v]++
	}

	var res []string

	for k, v := range d1 {
		if v == 1 {
			if _, ok := d2[k]; !ok {
				res = append(res, k)
			}
		}
	}

	for k, v := range d2 {
		if v == 1 {
			if _, ok := d1[k]; !ok {
				res = append(res, k)
			}
		}
	}

	return res 
}
